### PR TITLE
Update SMS Notification Class

### PR DESCRIPTION
**Pull request by michaelwehrley**
* Refactor SMS notification and add spec
* Add host applications' lesson notification rake task
* Update rake task to consider app name.

[#105006670]
[#105229798]

### CODE REVIEW

File: app/models/delivered_message.rb [hunk 1]
```diff
@@ -43,4 +47,13 @@ def deliver_notifications
   rescue
```
Comment (ericcf) [inline app/models/delivered_message.rb]: This seems like a good spot to notify Sentry with Raven. What do you think?

File: app/models/message_sms_notification.rb [hunk 2]
```diff
@@ -1,29 +1,44 @@
...
+  def initialize(body:, phone_number:)
+    @body = body
+    @phone_number = phone_number
   end
 
-  def self.deliver(message)
-    if Rails.env.staging? || Rails.env.production?
+  def deliver
+    if environment.staging? || environment.production?
       client = Twilio::REST::Client.new(
```
Comment (ericcf) [inline app/models/message_sms_notification.rb]: Would this be worth extracting to the following?
```ruby
def sms_client
  @sms_client ||= Twilio::REST::Client.new(
    config.twilio_account_sid,
    config.twilio_auth_token
  )
end
```

File: app/models/message_sms_notification.rb [hunk 3]
```diff
@@ -1,29 +1,44 @@
...
+    Rails.env
+  end
+
+  def from_telephone_number
+    config.try(:twilio_account_telephone_number)
+  end
+
+  def message
+    {
+      to: "+#{ phone_number }",
```
Comment (ericcf) [inline app/models/message_sms_notification.rb]: Any reason not to prepend with `+1` so that clinicians just have to enter a 10 digit standard number?

Comment (michaelwehrley) [inline app/models/message_sms_notification.rb]: good idea.  Thanks!

File: app/models/delivered_message.rb (hunk 1 above)
Comment (michaelwehrley) [inline app/models/delivered_message.rb]: :smile:

File: app/models/message_sms_notification.rb (hunk 2 above)
Comment (michaelwehrley) [inline app/models/message_sms_notification.rb]: Good catch.

File: app/models/message_sms_notification.rb [hunk 4]
```diff
@@ -1,29 +1,47 @@
...
-      client = Twilio::REST::Client.new(
-        Rails.application.config.twilio_account_sid,
-        Rails.application.config.twilio_auth_token
-      )
+  def deliver
+    if environment.staging? || environment.production?
+      sms_client
 
-      client.account.messages.create message
+      @sms_client.account.messages.create message
```
Comment (ericcf) [inline app/models/message_sms_notification.rb]: `sms_client.account.messages.create message`

File: app/models/message_sms_notification.rb [hunk 5]
```diff
@@ -1,29 +1,47 @@
...
+    Rails.env
+  end
+
+  def from_telephone_number
+    config.try(:twilio_account_telephone_number)
+  end
+
+  def message
+    {
+      to: "+1#{ phone_number }",
```
Comment (ericcf) [inline app/models/message_sms_notification.rb]: :+1:

**Pull request merged**